File: LeaveManagementBackend/LeaveManagementSystem.Models/Models/Leave.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LeaveManagementSystem.Models.Models
{
    public class Leave
    {
        [Key]
        public int Id { get; set; }

        [ForeignKey("User")]
        public int UserId { get; set; }
        public User User { get; set; }
        [ForeignKey("LeaveType")]
        public int LeaveTypeId { get; set; }
        public LeaveType LeaveType { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public DateTime DateOfRequest { get; set;}
        public string ReasonForLeave { get; set; }
        public string Status { get; set; }

    }
}

File: LeaveManagementFrontend/src/app/interfaces/leave.ts
export interface ILeave {
    id:number,
    userId:number,
    leaveTypeId:number,
    startDate:Date,
    endDate:Date,
    dateOfRequest:Date,
    reasonForLeave:string,
    status:string,
    user:{
        firstName:string
    },
    leaveType:{
        type:string
    }
}
